### ISSUE TITLE
Support for additional languages in the Dashboard

### ISSUE THREAD

**Issue by LiteSun** [apache/apisix-dashboard#1407]
# Feature request
## Please describe your feature
Currently, apisix-dashboard already supports switching between Chinese and English. We need more language support to allow more people from different regions to use our dashboard.
- [x] Chinese
- [x] English 
- [ ] Japanese
- [ ] German
- [ ] ...

Related links on completing i18n development：https://github.com/apache/apisix-dashboard/blob/a34317193cece2532ed333aaaacfd3a6af7328c4/docs/I18N_USER_GUIDE.md

Welcome your contribution!

**Comment by vinayaksh42** [comment]
Does this issue require Japanese and german language prior knowledge?

**Comment by juzhiyuan** [comment]
Ya! @vinayaksh42 Would u like to contribute to Japanese?

**Comment by vinayaksh42** [comment]
@vinayaksh42 I don't know Japanese, I do know Hindi(India) if you ever wanted to add it?

**Comment by juzhiyuan** [comment]
All languages are welcome! And we do welcome you to maintain them :)

**Comment by stu01509** [comment]
Hi @juzhiyuan, @LiteSun 

Can I add the Traditional Chinese?

**Comment by membphis** [comment]
ping @Junnplus @LiteSun

**Comment by juzhiyuan** [comment]
@stu01509 ya, of course!

**Comment by stu01509** [comment]
Hi @juzhiyuan 

Can I create the **zh-TW.ts** and **zh-TW** folder in the locales folder right?

**Comment by juzhiyuan** [comment]
🤔 This needs @qier222 's confirmation, not sure if it supports this.

**Comment by qier222** [comment]
> Hi @juzhiyuan
> 
> Can I create the **zh-TW.ts** and **zh-TW** folder in the locales folder right?

Yes, you can. Maybe we should use `zh-tw` instead of `zh-TW` since it will appear on the website URL?

**Comment by juzhiyuan** [comment]
ya, due to the Dashboard is still in progress and its documentation will get updated more frequently, how about involing Apache APISIX's documentation?

**Comment by anldrms** [comment]
> # Feature request
> ## Please describe your feature
> Currently, apisix-dashboard already supports switching between Chinese and English. We need more language support to allow more people from different regions to use our dashboard.
> 
> * [x]  Chinese
> * [x]  English
> * [ ]  Japanese
> * [ ]  German
> * [ ]  ...
> 
> Related links on completing i18n development：https://github.com/apache/apisix-dashboard/blob/a34317193cece2532ed333aaaacfd3a6af7328c4/docs/I18N_USER_GUIDE.md
> 
> Welcome your contribution!

I'm currently working on Turkish translation.

**Comment by Bhanu-code** [comment]
I can to this by adding google translate, Please assign me this issue.

**Comment by Bhanu-code** [comment]
thank you for assigning me, soon I will do it.

**Comment by Bhanu-code** [comment]
How to join the ASF slack channel? please somebody invite me.

**Comment by Bhanu-code** [comment]
I'm not getting enough time to work on this issue, so I am unassigning to not be stale,. Others can ask to be assigned.

**Comment by bzp2010** [comment]
This has been implemented in legacy versions 2.x and 3.x. It will also be implemented in the next post-renovation stage, which is planned. #2981

**Comment by SkyeYoung** [comment]
We have added full English support in #3000, welcome to help us provide more languages. (Better to wait for a while, as I expect to provide an optimization to reduce the number of i18n keys that need to be maintained)

**Comment by SkyeYoung** [comment]
The optimization has been completed. Contributions are welcome.

**Comment by ajinkya22-dev** [comment]
Can I add Spanish (es) translations for common.json to the locales directory?

**Comment by anldrms** [comment]
Hey @juzhiyuan , i can help you guys to improve Turkish translation let me know if you need help.
